File: app/src/main/java/com/comicsopentrends/rest/AuthInterceptor.java
package com.comicsopentrends.rest;


import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

class AuthInterceptor implements Interceptor {

    private final String tokenClans;

    AuthInterceptor(String tokenClans) {
        this.tokenClans = tokenClans;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {

        Request newRequest = chain.request().newBuilder()
                .addHeader("Authorization", "Bearer " + tokenClans)
                .build();
        return chain.proceed(newRequest);
    }
}

File: app/src/main/java/com/comicsopentrends/rest/AuthInterceptor.kt
package com.comicsopentrends.rest


import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

internal class AuthInterceptor(private val tokenClans: String) : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {

        val newRequest = chain.request().newBuilder()
                .addHeader("Authorization", "Bearer $tokenClans")
                .build()
        return chain.proceed(newRequest)
    }
}